tests/unit/lms/data_tasks/report/create_from_scratch/01_functions/test_01_date_functions.py

```diff
@@ -0,0 +1,46 @@
+from datetime import datetime
+
+import importlib_resources
+import pytest
+from data_tasks.sql_script import SQLScript
+
+TASK_ROOT = importlib_resources.files("lms.data_tasks")
+
+
+class TestDateFunctions:
+    ONE_YEAR_AGO = datetime.now().replace(year=datetime.now().year - 1).date()
+    TWO_YEARS_AGO = datetime.now().replace(year=datetime.now().year - 2).date()
+
+    @pytest.mark.usefixtures("with_date_functions")
+    @pytest.mark.parametrize(
+        "timescale,value,expected",
+        (
+            ("trailing_year", "NOW()", ONE_YEAR_AGO),
+            # Technically this test could fail if you run this exactly at midnight
+            ("trailing_year", "NOW() - INTERVAL '1 second'", ONE_YEAR_AGO),
+            ("trailing_year", "NOW() - INTERVAL '1 day'", ONE_YEAR_AGO),
+            ("trailing_year", "NOW() - INTERVAL '1 year'", TWO_YEARS_AGO),
+            ("trailing_year", "NOW() - INTERVAL '1 year 1 day'", TWO_YEARS_AGO),
+        ),
+    )
+    def test_multi_truncate(self, db_session, timescale, value, expected):
+        row = db_session.execute(
+            f"SELECT report.multi_truncate('{timescale}', ({value})::DATE)"
+        ).one_or_none()
+
+        assert row[0] == expected
```

Comment:
This was at the point where I wasn't sure it was working as it needs to. There are a lot of easy mistakes to make this this and I got it wrong a lot of times.